### PR TITLE
Swift

### PR DESCRIPTION
**Pull request by ypawar159**
This is completely different approach. Using per-built libraries wherever possible.
Using xbmcswift2 micro-framework for useful functions and CLI DEBUGGING.
Fully functional Download Feature implemented.
Let me know your thoughts on this.

### CODE REVIEW

File: resources/settings.xml [hunk 1]
```diff
@@ -12,4 +12,10 @@
 		<setting id="prefres" label="Prefere Resolution" type="labelenum" values="360p|480p|720p|1080p" default="1080p"/>
 		<setting id="autoplay" label="Automatically play first result" type="bool" default="false"/>
 	</category>
+	<category label="Aria2">
```
Comment (DxCx) [inline resources/settings.xml]: can you please give them some `aria_` prefix to ids?

File: resources/lib/ui/embed_extractor.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def playSource():
         reqObj = http.send_request(playUrl)
         return VIDEO_RE.findall(reqObj.text)[0]
 
-    return playSource
+    return playSource()
```
Comment (DxCx) [inline resources/lib/ui/embed_extractor.py]: commented about this like in 3 diff PRs...
you sholud NOT call playsource at this stage, when you play the url, and it's a function, it gets called..
this is the way we "lazy-load" episodes, and actualy resovle them only after selection.. much less load time.

File: resources/lib/ui/control.py [hunk 3]
```diff
@@ -104,3 +120,37 @@ def draw_items(video_data):
             xbmc_add_player_item(vid['name'], vid['url'], vid['image'])
     xbmcplugin.endOfDirectory(HANDLE, succeeded=True, updateListing=False, cacheToDisc=True)
     return True
+
```
Comment (DxCx) [inline resources/lib/ui/control.py]: aren't those leftovers now that you integrated xbmcswift?

File: downloadmanager.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                    plugin.log.info(end)
+                    plugin.log.info(eps)
+                    fetch_downloads(plugin, eps)
+
+
+def fetch_downloads(plugin, episode_list):
+    preferred_resolution = int(plugin.get_setting("prefres")[:-1])
+    downloads = []
+    for episode in episode_list:
+        fetched = fetch_sources([('RapidVideo', episode["source"])], MockedDialog(), True)
```
Comment (DxCx) [inline downloadmanager.py]: why forced to RapidVideo??? :O

File: default.py [hunk 5]
```diff
@@ -25,139 +23,309 @@
...
+    serverChoice = filter(lambda x: plugin.get_setting(x[0]) == 'true', SERVER_CHOICES.iteritems())
+    serverChoice = map(lambda x: x[1], serverChoice)
+    sources = filter(lambda x: x[0] in serverChoice, sources)
+    autoplay = True if 'true' in plugin.get_setting('autoplay') else False
     s = SourcesList(sources, autoplay, sortResultsByRes, {
-        'title': control.lang(30100),
-        'processing': control.lang(30101),
-        'choose': control.lang(30102),
-        'notfound': control.lang(30103),
+        'title': "Fetching Sources",
```
Comment (DxCx) [inline default.py]: please use strings from string table..
this plugin should support multi-langugue

File: default.py [hunk 6]
```diff
@@ -25,139 +23,309 @@
...
-    serverChoice = map(lambda x: x[1], serverChoice)
-    sources = filter(lambda x: x[0] in serverChoice, sources)
+    prefereResSetting = utils.parse_resolution_of_source(plugin.get_setting('prefres'))
+    filtered_urls = filter(lambda x: utils.parse_resolution_of_source(x[0]) <= prefereResSetting, fetched_urls)
+    return sorted(filtered_urls, key=lambda x: utils.parse_resolution_of_source(x[0]), reverse=True)
+
 
-    autoplay = True if 'true' in control.getSetting('autoplay') else False
+MENU_ITEMS = [
+    ("Latest", "latest_d"),
```
Comment (DxCx) [inline default.py]: also here, you removed support or string table..

**Comment by ypawar159** [comment]
Hi DxCx,
I will try to implement all the suggestions.
Will take some time.
And xbmcswift2 is installed through pip during development.
And available in kodi repositories so no need to package it with the plugin.
But after implementing this swift approach I later discovered that the original author has abandoned the project. So need to take this in consideration for moving forward with this approach. There are alternative libraries which do small parts, like proper url routing, other helper functions. So I would like to discuss what would be best approach so that things won't break in future.

File: default.py (hunk 6 above)
Comment (ypawar159) [reply default.py]: I kind of didn't understand how to implement it with new framework. Will try to solve it.

File: downloadmanager.py (hunk 4 above)
Comment (ypawar159) [reply downloadmanager.py]: That's the only stable source I could test at my setup.

**Pull request closed without merging**